src/vsi.rs

```diff
@@ -28,7 +28,14 @@ fn _read_dir(path: &Path, recursive: bool) -> Result<Vec<String>> {
         data
     };
 
-    Ok(_string_array(data))
+    let strings = _string_array(data);
```

Comment:
We can do it in a later PR, but you could add `_pathbuf` and `_pathbuf_array` functions to avoid the reallocation here.

Reply:
It's getting late here. Let me do that tomorrow.